ecosystem/indexer-grpc/indexer-grpc-integration-test-transaction-generator/src/transaction_generator.rs

```diff
@@ -0,0 +1,273 @@
+    pub fn get_transaction_generator(self) -> TransactionGenerator {
+        TransactionGenerator::new(
+            self.test_cases_folder,
+            self.output_test_cases_folder,
+            self.aptos_node_binary,
+            self.node_version,
+            self.node_config,
+        )
+    }
+}
+
+/// Struct that generates transactions for testing purposes.
+/// Internally, it brings up a local node and sends transactions based on the test case.
+#[derive(Debug)]
+pub struct TransactionGenerator {
+    /// The local node that the transaction generator uses to send transactions.
+    // node: Node,
+    /// The test case that the transaction generator uses to generate transactions.
+    // test_case: TestCaseConfig,
+    test_cases_folder: PathBuf,
+
+    /// The folder where the generated test cases will be stored.
+    output_test_cases_folder: PathBuf,
+
+    /// Override node config path.
+    node_config: Option<PathBuf>,
+
+    /// Whether the transaction generator has been initialized correctly.
+    is_initialized: bool,
+
+    /// The aptos node binary path. This is used to start the local node.
+    aptos_node_binary: PathBuf,
+
+    /// The process handle of the local node.
+    node_process: Option<Child>,
+
+    /// Release version: `1.13.0`` or `latest`.
+    node_version: String,
+
+    /// Move Executor.
+    move_executor: Option<MoveExecutor>,
+}
+
+impl TransactionGenerator {
+    /// Create a new transaction generator.
+    /// The transaction generator is not initialized until `initialize` is called.
+    fn new(
+        test_cases_folder: PathBuf,
+        output_test_cases_folder: PathBuf,
+        aptos_node_binary: PathBuf,
+        node_version: String,
+        node_config: Option<PathBuf>,
+    ) -> Self {
+        Self {
+            test_cases_folder,
+            output_test_cases_folder,
+            node_config,
+            is_initialized: false,
+            node_process: None,
+            aptos_node_binary,
+            node_version,
+            move_executor: None,
+        }
+    }
+
+    /// Initialize the transaction executor, which includes:
+    ///  - A local node.
+    ///  - MoveExecutor.
+    pub async fn initialize(&mut self) -> anyhow::Result<()> {
+        // Start the local node.
+        self.node_process =
+            Some(start_localnode(self.aptos_node_binary.clone(), self.node_config.clone()).await?);
+        // Create the Move executor.
+        let move_executor = MoveExecutor::new(DEFAULT_LOCAL_NODE_API_URL)
+            .context("Failed to create Move executor.")?;
+        self.move_executor = Some(move_executor);
+        self.is_initialized = true;
+        Ok(())
+    }
+
+    /// Build the transactions based on the test cases read.
+    /// TODO: refactor this.
+    pub async fn build_test_cases(&self) -> anyhow::Result<()> {
+        if !self.is_initialized {
+            return Err(anyhow::anyhow!("Transaction generator is not initialized."));
+        }
```

Comment:
I don't like this pattern, it pushes something that could be guaranteed at compile time to be a runtime check. This way, the user has to remember to call `initialize`. Can we just delete the `initialize` function and move all of that stuff into `new`. Then we don't need the `is_initialized` var.